apps/src/javalab/Theater.js

```diff
@@ -7,25 +7,41 @@ export default class Theater {
     this.context = null;
     this.onOutputMessage = onOutputMessage;
     this.onNewlineMessage = onNewlineMessage;
+    this.loadEventsFinished = 0;
   }
 
   handleSignal(data) {
     switch (data.value) {
       case TheaterSignalType.AUDIO_URL: {
+        // Wait for the audio to load before starting playback
         this.getAudioElement().src = data.detail.url;
+        this.getAudioElement().oncanplaythrough = () => this.startPlayback();
         break;
       }
       case TheaterSignalType.VISUAL_URL: {
+        // Preload the image. Once it's ready, start the playback
         this.getImgElement().src = data.detail.url;
+        this.getImgElement().onload = () => this.startPlayback();
         break;
       }
       default:
         break;
     }
   }
 
+  startPlayback() {
+    this.loadEventsFinished++;
+    // We expect exactly 2 responses from Javabuilder. One for audio and one for video.
+    // Wait for both to respond and load before starting playback.
+    if (this.loadEventsFinished > 1) {
+      this.getImgElement().style.visibility = 'visible';
```

Comment:
Curious, does toggling the visibility fix the broken image issue too?

Reply:
Yes, it does! Good eye. Thanks for the reminder too - I forgot to include that in the description :)